cache/manager.go

```diff
@@ -740,6 +761,122 @@ func (cm *cacheManager) Merge(ctx context.Context, inputParents []ImmutableRef,
 	return rec.ref(true, dhs), nil
 }
 
+func (cm *cacheManager) Diff(ctx context.Context, lower, upper ImmutableRef, opts ...RefOption) (ir ImmutableRef, rerr error) {
+	if lower == nil && upper == nil {
```

Comment:
I tend to think these checks are better at ops level? In here they just look like invalid.